engine/src/main/java/org/hibernate/search/engine/mapper/mapping/building/impl/AbstractIndexBindingContext.java

```diff
@@ -66,38 +67,53 @@ public IndexSchemaElement getSchemaElement(IndexSchemaContributionListener liste
 		return new IndexSchemaElementImpl<>(
 				getTypeFactory(),
 				indexSchemaObjectNodeBuilder,
-				new NotifyingNestingContext( nestingContext, listener )
+				new NotifyingNestingContext( nestingContext, listener ),
+				isMultiValuedFromParentNotPropagated()
 		);
 	}
 
 	@Override
-	public Optional<IndexedEmbeddedBindingContext> addIndexedEmbeddedIfIncluded(MappableTypeModel parentTypeModel,
+	public Optional<IndexedEmbeddedBindingContext> addIndexedEmbeddedIfIncluded(
+			MappableTypeModel parentTypeModel, boolean multiValued,
 			String relativePrefix, ObjectFieldStorage storage, Integer maxDepth, Set<String> includePaths) {
 		return nestingContext.addIndexedEmbeddedIfIncluded(
 				parentTypeModel, relativePrefix, maxDepth, includePaths,
-				new NestedContextBuilderImpl( indexSchemaRootNodeBuilder, indexSchemaObjectNodeBuilder, storage )
+				new NestedContextBuilderImpl(
+						indexSchemaRootNodeBuilder, indexSchemaObjectNodeBuilder, storage,
+						isMultiValuedFromParentNotPropagated() || multiValued
```

Comment:
Ok, so the code is right, but the name is very, very bad indeed. When`isMultiValuedFromParentNotPropagated` is `true`, it actually means the parent is multi-valued, but that information could not be propagated to the index and thus should be propagated in children.

Think of a `@IndexedEmbedded` property of type `List<MyEmbedded>` which doesn't add any object field: its fields will automatically become multi-valued, because the property has multiple `MyEmbedded` values and each `MyEmbedded` value has to be translated into field values. So even if `MyEmbedded` only declares single-valued fields, in the end these fields will become multi-valued in the `@IndexedEmbedded`.

I'll try to find a better name for `isMultiValuedFromParentNotPropagated`.